src/sentry/monitors/consumers/check_in.py

```diff
@@ -184,6 +184,7 @@ def _process_message(wrapper: Dict) -> None:
                     status=status,
                     date_added=date_added,
                     date_updated=start_time,
+                    config_data={"expected_time": monitor_environment.next_checkin},
```

Comment:
@davidenwang should we actually store this as the monitor config + the "exact" `expected_time`.

So basically ignore this logic and just us `get_next_schedule`

https://github.com/getsentry/sentry/blob/1ab1775fd5d15074f64d48579f5e960e92290c39/src/sentry/monitors/models.py#L252-L255

(We probably want to change up the code a bit to do this cleanly)

Reply:
makes sense i'm down for that